src/foam/nanos/auth/ServiceProvider.js

```diff
@@ -52,8 +52,16 @@ foam.CLASS({
     },
     {
       name: 'permissionsGranted',
-      transient: true,
-      javaGetter: 'return new String[] { "serviceprovider.read." + getId() };',
+      networkTransient: true,
+      javaGetter: `
```

Comment:
another idea to use `javaFactory: 'return new String[] { "serviceprovider.read" + getId() };'` instead then add "serviceprovider.read.treviso" permission to the treviso spid.

Reply:
Wouldn't the extra journal entry be a bit redundant if there's already logic in here to do that?

Reply:
no, javaFactory is only for when the service provider doesn't have granted permission.

Reply:
What I mean is `"serviceprovider.read" + getId()` is already in the code somewhere, so I don't think it's good to expect that someone adding a ServiceProvider entry to the journal to know they have to add `serviceprovider.read.<id>` if and only if they happen to override the property. (or in other words, I would never expect adding `permissionsGranted: []` to a capability to remove a permission that's implied by the subclass)

Reply:
Agreed but wonder do we have the case where someone else other than the admin is allowed to modify the granted permission on a service provider? We need a way to make the "serviceprovider.read.<id>" permission inherent and permanent to a service provider.

Reply:
I believe no user will be able to remove that since it's always included by the getter. Although if users can edit these permissions then I should probably take out `networkTransient` too right?